temporian/implementation/numpy/operators/calendar_day_of_month.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict
+from datetime import datetime
+
+import numpy as np
+from temporian.core.operators.calendar_day_of_month import (
+    CalendarDayOfMonthOperator,
+)
+from temporian.implementation.numpy.data.event import NumpyEvent
+from temporian.implementation.numpy.data.event import NumpyFeature
+
+
+class CalendarDayOfMonthNumpyImplementation:
+    def __init__(self, operator: CalendarDayOfMonthOperator) -> None:
+        super().__init__()
+        self.operator = operator
+
+    def __call__(self, event: NumpyEvent) -> Dict[str, NumpyEvent]:
+        data = {}
+        for index, timestamps in event.sampling.data.items():
+            days = np.array(
+                [datetime.fromtimestamp(ts).day for ts in timestamps]
```

Comment:
Currently, If the user sends a Date Object we convert the timestamp to a UTC epoch, so the timezone for every timestamp in the sampling will be UTC.

Reply:
Nice catch! Was planning on adding some edge case tests (e.g. at 00:01 and 23:59) to ensure exactly this, will do so and fix it. Agree with adding the timezone as an optional arg.

Reply:
from the link Mathieu posted:
> If tz is not None, it must be an instance of a [tzinfo](https://docs.python.org/3/library/datetime.html#datetime.tzinfo) subclass

skipping timezone argument for now since we only support numeric/string arguments for operators. creating issue off of this conversation for the future.